var/spack/repos/builtin/packages/qwt/package.py

```diff
@@ -32,11 +34,19 @@ class Qwt(QMakePackage):
     patch("no-designer.patch", when="~designer")
     patch("no-opengl_6_1.patch", when="@6.1 ~opengl")
 
-    depends_on("qt+tools", when="+designer")
-    depends_on("qt+opengl", when="+opengl")
-
-    # Qwt does not support Qt6; this picks the right qmake provider
-    conflicts("^qt-base", msg="Qwt requires Qt5")
+    depends_on("qmake")
+    with when("^[virtuals=qmake] qt-base"):
+        depends_on("qt-svg")
+        depends_on("qt-tools", when="+designer")
+        depends_on("qt-base+opengl", when="+opengl")
```

Comment:
This should prevent attempts to build `qwt ^qt-base~widgets` which AFAICS failed for me with:
```py
     16    Project ERROR: Unknown module(s) in QT: openglwidgets opengl
```
```suggestion
        depends_on("qt-base+opengl+widgets", when="+opengl")
```